Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.DataGridViewRowCollection.cs

```diff
@@ -0,0 +1,614 @@
+        // </summary>
+        // 
+        // <returns>
+        // true if the <see cref="T:System.Windows.Forms.DataGridViewRow"/> is in the <see cref="T:System.Windows.Forms.DataGridViewRowCollection"/>; otherwise, false.
+        // </returns>
+        // <param name="dataGridViewRow">The <see cref="T:System.Windows.Forms.DataGridViewRow"/> to locate in the <see cref="T:System.Windows.Forms.DataGridViewRowCollection"/>.</param>
+        // public virtual bool Contains(DataGridViewRow dataGridViewRow)
+        
+        /// <summary>
+        /// Copies the items from the collection into the specified <see cref="T:System.Windows.Forms.DataGridViewRow"/> array, starting at the specified index.
+        /// </summary>
+        /// <param name="array">A <see cref="T:System.Windows.Forms.DataGridViewRow"/> array that is the destination of the items copied from the <see cref="T:System.Windows.Forms.DataGridViewRowCollection"/>.</param><param name="index">The zero-based index in <paramref name="array"/> at which copying begins.</param><exception cref="T:System.ArgumentNullException"><paramref name="array"/> is null. </exception><exception cref="T:System.ArgumentOutOfRangeException"><paramref name="index"/> is less than zero. </exception><exception cref="T:System.ArgumentException"><paramref name="array"/> is multidimensional.-or- The number of elements in the <see cref="T:System.Windows.Forms.DataGridViewRowCollection"/> is greater than the available space from <paramref name="index"/> to the end of <paramref name="array"/>. </exception>
+        public void CopyTo(DataGridViewRow[] array, int index)
+        {
+            Contract.Requires(array != null);
+            Contract.Requires(index >= 0 && index <= this.Count);
+        }
+        
+        /// <summary>
+        /// Returns the index of the first <see cref="T:System.Windows.Forms.DataGridViewRow"/> that meets the specified criteria.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The index of the first <see cref="T:System.Windows.Forms.DataGridViewRow"/> that has the attributes specified by <paramref name="includeFilter"/>; -1 if no row is found.
+        /// </returns>
+        /// <param name="includeFilter">A bitwise combination of <see cref="T:System.Windows.Forms.DataGridViewElementStates"/> values.</param><exception cref="T:System.ArgumentException"><paramref name="includeFilter"/> is not a valid bitwise combination of <see cref="T:System.Windows.Forms.DataGridViewElementStates"/> values.</exception>
+        public int GetFirstRow(DataGridViewElementStates includeFilter)
+        {
+            Contract.Requires(
+                !((includeFilter &
+                  ~(DataGridViewElementStates.Displayed | DataGridViewElementStates.Frozen |
+                    DataGridViewElementStates.ReadOnly | DataGridViewElementStates.Resizable |
+                    DataGridViewElementStates.Selected | DataGridViewElementStates.Visible)) !=
```

Comment:
It should be easy to invert this check. Just remove first `!` and use `== None`.